modules/agar/src/main/ts/ephox/agar/mouse/Clicks.ts

```diff
@@ -75,22 +75,23 @@ const mouseUp = Fun.curry(event, 'mouseup');
 const mouseMove = Fun.curry(event, 'mousemove');
 const mouseOver = Fun.curry(event, 'mouseover');
 const mouseOut = Fun.curry(event, 'mouseout');
-const contextMenu = (settings: Settings) => event('contextmenu', { button: rightClickButton, ...settings });
+const contextMenu = (settings: Settings): (element: SugarElement<Node>) => void =>
+  event('contextmenu', { button: rightClickButton, ...settings });
 
 // Note: This can be used for phantomjs.
-const trigger = function (element: SugarElement<any>): any {
-  const ele: HTMLElement = element.dom;
+const trigger = function (element: SugarElement<HTMLElement>): void {
+  const ele = element.dom;
   if (ele.click !== undefined) {
     return ele.click();
   }
   // Adapted from: http://stackoverflow.com/questions/17468611/triggering-click-event-phantomjs
   point('click', leftClickButton, element, 0, 0);
-  return undefined;
+  return;
 };
 
-const point = (type: string, button: number, element: SugarElement<any>, x: number, y: number): void => {
+const point = (type: string, button: number, element: SugarElement<Node>, x: number, y: number): void => {
   // Adapted from: http://stackoverflow.com/questions/17468611/triggering-click-event-phantomjs
-  const ev: MouseEvent = (<Document> element.dom.ownerDocument).createEvent('MouseEvents');
+  const ev: MouseEvent = Traverse.owner(element).dom.createEvent('MouseEvents');
```

Comment:
Minor optimisation made here

Reply:
Optimisation in terms of types or minifying or...? A bit lost sorry

Reply:
In terms of code size and maintenance sorry. As it keeps all the document traversal logic in one place.